src/live-programming.rst

```diff
@@ -47,3 +47,210 @@ TODO: in music
 Programmable Physical Environments
 ==================================
 
+Overview
+--------
+
+Physical things includes nearly everything that will appear in life of human beings. When it comes to programmable physical things, the term should refer to a kind of physical system which might be a combination of software and hardware which can respond to the analog world. When we say 'environment', we focus more on the interactive function of this physical system with human beings. Now we can derive a definition for programmable physical environment, which is an interactively programmable physical system that can sense and respond to the real world.
```

Comment:
- We don't need to define "physical things" -- let's assume readers already know that
- Start this paragraph with "Programmable physical environments are..." (your definition at the end of this paragraph is good)